packages/sort_box/lib/sort_box.dart

```diff
@@ -5,40 +5,47 @@ typedef ComparableGetter<T> = Comparable Function(T);
 class SortBox<T extends Enum, R> {
   SortBox(
     this._properties,
-    this._secondaryBoxes,
+    this._boxes,
   );
 
   final Map<T, ComparableGetter<R>> _properties;
-  final Map<T, Box<T>> _secondaryBoxes;
+  final Map<T, Set<Box<T>>> _boxes;
 
-  List<R> sort(final List<R> input, final Box<T> box) {
+  List<R> sort(final List<R> input, final Box<T> box, [final Set<Box<T>>? presort]) {
```

Comment:
The changes make sense, but I feel like the interface gets way too complex. Could you please either add a lot of documentation on how it works or change the interface so that it is easier to understand?